--- libstdc++-v3/libsupc++/eh_alloc.cc.orig
+++ libstdc++-v3/libsupc++/eh_alloc.cc
@@ -111,13 +111,25 @@ namespace
   pool::pool()
     {
       // Allocate the arena - we could add a GLIBCXX_EH_ARENA_SIZE environment
       // to make this tunable.
       arena_size = (EMERGENCY_OBJ_SIZE * EMERGENCY_OBJ_COUNT
 		    + EMERGENCY_OBJ_COUNT * sizeof (__cxa_dependent_exception));
+
+      // --------- Patched by rdiez ---------
+      // Disable the emergency C++ exception memory pool in order to
+      // conserve memory in highly-constrained embedded environments.
+      if ( false )
+      {
       arena = (char *)malloc (arena_size);
+      }
+      else
+      {
+      arena = nullptr;
+      }
+
       if (!arena)
 	{
 	  // If the allocation failed go without an emergency pool.
 	  arena_size = 0;
 	  first_free_entry = NULL;
 	  return;
