--- newlib/libc/string/strerror.c.orig
+++ newlib/libc/string/strerror.c
@@ -390,12 +390,17 @@ _strerror_r (struct _reent *ptr,
 	int internal,
 	int *errptr)
 {
   char *error;
   extern char *_user_strerror (int, int, int *);
 
+  // --------- Patched by rdiez ---------
+  // Remove all strerror messages, because they take up a lot of space.
+  if ( 1 )
+    return "<n/a>";
+
   switch (errnum)
     {
     case 0:
       error = "Success";
       break;
 /* go32 defines EPERM as EACCES */
