--- newlib/libc/include/assert.h.orig	2012-10-16 21:00:30.000000000 +0200
+++ newlib/libc/include/assert.h	2014-10-03 16:03:45.000000000 +0200
@@ -13,8 +13,9 @@
 #ifdef NDEBUG           /* required by ANSI standard */
 # define assert(__e) ((void)0)
 #else
-# define assert(__e) ((__e) ? (void)0 : __assert_func (__FILE__, __LINE__, \
-						       __ASSERT_FUNC, #__e))
+# define assert(__e) ((__e) ? (void)0 : __assert_func ( __FILE__, __LINE__, \
+                                                        "<assert function names removed to reduce code size>" /* __ASSERT_FUNC */, \
+                                                        "<assert expressions removed to reduce code size>" /* #__e */ ))
 
 # ifndef __ASSERT_FUNC
   /* Use g++'s demangled names in C++.  */
