--- newlib/libc/string/strerror.c.old	2015-07-25 10:51:12.090137911 +0200
+++ newlib/libc/string/strerror.c	2015-07-25 10:52:59.298138172 +0200
@@ -389,6 +389,13 @@ _DEFUN (_strerror_r, (ptr, errnum, inter
   char *error;
   extern char *_user_strerror _PARAMS ((int, int, int *));
 
+
+  // --------- Patched by rdiez ---------
+  // Remove all strerror messages, because they take up a lot of space.
+  if ( 1 )
+    return "<n/a>";
+
+
   switch (errnum)
     {
     case 0:
